rust/src/countme/cookie.rs

```diff
@@ -74,13 +66,11 @@ impl CookieV0 {
         Ok(c)
     }
 
-    fn persist(epoch: i64, window: i64) -> Result<()> {
-        let c = CookieV0 { epoch, window };
-        let state = openat::Dir::open(STATE_DIR)?;
-        let mut fw = state.new_file_writer(COUNTME_COOKIE, 0o644)?;
-        fw.writer.write_all(serde_json::to_string(&c)?.as_bytes())?;
-        fw.complete()?;
-        Ok(())
+    fn persist(epoch: i64, window: i64) -> Result<String> {
```

Comment:
Hmm this stopped writing to disk?  I'd call it `to_json()` then probably.

But I don't see the calling code of `persist()` actually writing to disk?

Reply:
Looks like that's in `Cookie::persist`. (I was confused too by the `CookieV0` vs `Cookie` separation.)

Reply:
It is indeed split to mirror DNF versioned cookie semantics and enable keeping the state in a single structure, keeping the on-disk format as a temporary structure only.